From fa572d93cfa463f61432dd92239e747f9642fbae Mon Sep 17 00:00:00 2001
From: David Kahles <david.kahles96@gmail.com>
Date: Fri, 12 Jan 2018 15:23:53 +0100
Subject: Fix token page URL

Summary:
It seems like Google changed the URL of the token page.
Maybe we should use QString::startsWith, to be futureproof?

BUG: 388483

Test Plan:
- Login in KOrganizer and KMail works again
- All tests pass (though i think there is no oauth2 test)

Reviewers: dvratil, mlaurent

Reviewed By: mlaurent

Differential Revision: https://phabricator.kde.org/D9843
---
 src/core/ui/authwidget_p.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/ui/authwidget_p.h b/src/core/ui/authwidget_p.h
index 78b0e7f..29ece44 100644
--- a/src/core/ui/authwidget_p.h
+++ b/src/core/ui/authwidget_p.h
@@ -79,7 +79,7 @@ class Q_DECL_HIDDEN AuthWidgetPrivate: public QObject {
     bool isSigninPage(const QUrl &url) const { return url.path() == QLatin1String("/signin/oauth"); }
     bool isUsernameFrame(const QUrl &url) { return url.path() == QLatin1String("/signin/oauth/identifier"); }
     bool isPasswordFrame(const QUrl &url) { return url.path() == QLatin1String("/signin/v2/challenge/pwd"); }
-    bool isTokenPage(const QUrl &url) { return url.path() == QLatin1String("/o/oauth2/approval/v2"); }
+    bool isTokenPage(const QUrl &url) { return url.path() == QLatin1String("/o/oauth2/approval/v2/approvalnativeapp"); }
 
     void setSslIcon(const QString &icon);
 
-- 
cgit v0.11.2

